hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/V2MessageUtil.java

```diff
@@ -82,6 +82,7 @@ public static String getSecurity(String v2Message) {
 	private static String getMSHSegment(String v2Message) {
 		String mshSegment = "";
 		if (!StringUtils.isBlank(v2Message)) {
+			v2Message = StringUtils.startsWith(v2Message, V2MessageUtil.SegmentType.MSH.toString()) ? v2Message : v2Message.substring(8);
```

Comment:
In general it's best practice not to modify a method argument directly as it can be confusing. Best to create a local variable (e.g. trimmedV2Message, adjustedV2Message).